clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -61,73 +83,250 @@ public void setup() {
+        assertEquals(0, commitManager.pendingRequests.inflightOffsetFetches.size());
+    }
+
+    @Test
+    public void testPoll_EnsureEmptyPendingRequestAfterPoll() {
         CommitRequestManager commitRequestManger = create(true, 100);
-        time.sleep(100);
-        commitRequestManger.clientPoll(time.milliseconds());
-        NetworkClientDelegate.PollResult res = commitRequestManger.poll(time.milliseconds());
-        time.sleep(100);
-        // We want to make sure we don't resend autocommit if the previous request has not been completed
-        assertEquals(Long.MAX_VALUE, commitRequestManger.poll(time.milliseconds()).timeUntilNextPollMs);
+        commitRequestManger.addOffsetCommitRequest(new HashMap<>());
+        assertEquals(1, commitRequestManger.unsentOffsetCommitRequests().size());
+        commitRequestManger.poll(time.milliseconds());
+        assertTrue(commitRequestManger.unsentOffsetCommitRequests().isEmpty());
+        assertEmptyPendingRequests(commitRequestManger);
+    }
 
+    @Test
+    public void testAutocommit_ResendAutocommitAfterException() {
+        CommitRequestManager commitRequestManger = create(true, 100);
+        time.sleep(100);
+        commitRequestManger.updateAutoCommitTimer(time.milliseconds());
+        List<CompletableFuture<ClientResponse>> futures = assertPoll(1, commitRequestManger);
+        time.sleep(99);
         // complete the autocommit request (exceptionally)
-        res.unsentRequests.get(0).future().completeExceptionally(new KafkaException("test exception"));
+        futures.get(0).completeExceptionally(new KafkaException("test exception"));
 
         // we can then autocommit again
-        commitRequestManger.clientPoll(time.milliseconds());
-        res = commitRequestManger.poll(time.milliseconds());
-        assertEquals(1, res.unsentRequests.size());
+        commitRequestManger.updateAutoCommitTimer(time.milliseconds());
+        assertPoll(0, commitRequestManger);
+        time.sleep(1);
+        commitRequestManger.updateAutoCommitTimer(time.milliseconds());
+        assertPoll(1, commitRequestManger);
+        assertEmptyPendingRequests(commitRequestManger);
     }
 
     @Test
-    public void testEnsureStagedCommitsPurgedAfterPoll() {
+    public void testAutocommit_EnsureOnlyOneInflightRequest() {
         CommitRequestManager commitRequestManger = create(true, 100);
-        commitRequestManger.add(new HashMap<>());
-        assertEquals(1, commitRequestManger.stagedCommits().size());
-        NetworkClientDelegate.PollResult res = commitRequestManger.poll(time.milliseconds());
-        assertTrue(commitRequestManger.stagedCommits().isEmpty());
+        time.sleep(100);
+        commitRequestManger.updateAutoCommitTimer(time.milliseconds());
+        List<CompletableFuture<ClientResponse>> futures = assertPoll(1, commitRequestManger);
+        time.sleep(100);
+        commitRequestManger.updateAutoCommitTimer(time.milliseconds());
```

Comment:
@guozhangwang - Fixed setting the inflight request in the `maybeAutoCommit` method. This test should address this issue/comment.